parsers/rwcpopmelody_parser.py

```diff
@@ -0,0 +1,199 @@
+    m = midi.Pattern(tracks=melody, resolution=m.resolution,
+                     format=m.format, tick_relative=m.tick_relative)
+
+    # Write temporary midi file
+    temp_midi = out_file.replace(".jams", ".temp.mid")
+    midi.write_midifile(temp_midi, m)
+
+    # Load temp midi file using pretty_midi
+    pm = pretty_midi.PrettyMIDI(temp_midi)
+
+    assert len(pm.instruments) == 2
+    assert pm.instruments[0].notes == []
+    assert len(pm.instruments[1].notes) > 0
+
+    # Get the melody notes
+    notes = pm.instruments[1].notes
+
+    # Create jam
+    jam = jams.JAMS()
+
+    # Create annotation
+    midi_ann = jams.Annotation('pitch_midi')
+
+    # Add notes to the annotation
+    for note in notes:
+        midi_ann.append(time=note.start, duration=(note.end-note.start),
+                        value=note.pitch, confidence=1.)
+
+    # Fill annotation metadata
+    fill_annotation_metadata(midi_ann)
+
+    # Add annotation to jam
+    jam.annotations.append(midi_ann)
+
+    # Fill file metadata
+    n_track = int(os.path.basename(smf_file)[4:7]) - 1
+    fill_file_metadata(jam, metadata, n_track)
+
+    # Save JAMS
+    jam.save(out_file)
+
+    # Remove temporary midi file
+    os.remove(temp_midi)
+
+
+def process_folder(smf_dir, out_dir):
+    """Converts the original SMF annotations into the JAMS format (keeping only
+    the notes of the melody track), and saves them in the out_dir folder."""
+
+    # Collect all SMF annotations.
+    smf_files = jams.util.find_with_extension(smf_dir, '.MID', depth=1)
+
+    # Get metadata
+    metadata = pd.read_csv('data/rwcmelodypop/metadata.csv')
+
+    for smf in smf_files:
+
+        jams_file = (
+            os.path.join(out_dir,
+                         os.path.basename(smf).replace('.MID', '.jams')))
+        jams.util.smkdirs(os.path.split(jams_file)[0])
+        # Create a JAMS file for this track
+        print(smf)
+        create_jams(smf, jams_file, metadata)
+
+
+def main():
+    """Main function to convert the dataset into JAMS."""
+    parser = argparse.ArgumentParser(
+        description="Converts the RWC POP dataset to the JAMS format, keeping "
+                    "only the notes of the melody track.",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument("smf_dir",
+                        action="store",
+                        help="RWC POP SMF folder")
+    parser.add_argument("-o",
+                        action="store",
+                        dest="out_dir",
+                        default="RWCPOP_melody_jams",
+                        help="Output JAMS folder")
+    args = parser.parse_args()
+    start_time = time.time()
+
+    # Setup the logger
+    logging.basicConfig(format='%(asctime)s: %(message)s', level=logging.INFO)
+
+    # Run the parser
+    process_folder(args.smf_dir, args.out_dir)
+
+    # Done!
+    logging.info("Done! Took %.2f seconds.", time.time() - start_time)
```

Comment:
Why have a logger if it's only used for profiling elapsed time? Seems better to just remove it entirely and let external profilers measure main() if needed.